### ISSUE TITLE
Freezes whole desktop

### ISSUE THREAD

**Issue by hissssst** [flameshot-org/flameshot#2627]
### Flameshot Version

Flameshot v11.0.0 (-)
Compiled with Qt 5.15.3

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Nixos

### Description

Whole desktop gets frozen.
It's not a memory leak, because earlyoom doesn't kill the flameshot

### Steps to reproduce

1. just run it. Any configuration or options will do
2. During filechooser dialog it gets frozen

### System Information

Sway

**Comment by borgmanJeremy** [comment]
Have you followed the guide for setting up flameshot on sway? Do you have a notification manager like dunst installed and configured?

**Comment by hissssst** [comment]
Yeah, I have notification manager installed and configured

**Comment by borgmanJeremy** [comment]
When you say "just run it" what specifically are you doing? @nullobsi merged sway support and I believe it works for them.

**Comment by hissssst** [comment]
I run `flameshot gui`. It freezes when I run it from shell in VT in sway or when I run it from bind in sway

**Comment by hissssst** [comment]
I've just tested and this issue occurs even when flameshot started as a service and when not

**Comment by mmahmoudian** [comment]
Do you have a any big rise in IO?

**Comment by hissssst** [comment]
How do I check it?

**Comment by mmahmoudian** [comment]
To check IO you can use `iotop` for example.

**Comment by hissssst** [comment]
Yeah, but my system gets completely frozen, how do I record this

**Comment by mmahmoudian** [comment]
If you have IO problem, recording it is not a good idea as almost all screen recording software write to disk while recording, but if you want to give this a try, I can recommend installing peek.

**Comment by hissssst** [comment]
Yeah, I know that screen recording is not a good idea, perhaps you know some IO usage top which can just append to the tail of file and don't corrupt it when the system is restarted. Any suggestions?

**Comment by borgmanJeremy** [comment]
Actually a better route would be to run flameshot in a debugger and see where it gets stuck. I strongly suspect this is an issue with your notification daemon or d-bus on your pc.

**Comment by hissssst** [comment]
By the way, I don't think I have time to debug this using gdb. But I can share the VM image of my setup upon which you can perform tests

**Comment by MarkGotLasagna** [comment]
I'd like to help on resolving the same problem by providing the necessary info: I have the same issue.

I'm using `arch`, with `i3wm` as WM.

**Comment by mmahmoudian** [comment]
@MarkGotLasagna what is your Flameshot version, how many monitors do you have, with what configuration, how have you installed Flameshot (pacman, AUR, Flatpak, Snap, AppImage,...) how are you running Flameshot (one-off mode or daemon mode, `gui` or `screen` or `full`), when does it freeze (during starting, when you copy, when you are writing to disk,...)

**Comment by MarkGotLasagna** [comment]
- Version: `Flameshot v12.1.0 (-) Compiled with Qt 5.15.5`
- Monitors: 2  (eDP-i, HDMI-1)
- Configuration: default configuration (unchanged)
- Installed with: `pacman`
- Command: `flameshot gui`

The whole desktop freezes as I click on any interactive icon within the `flameshot gui` (can still `tty`). Using key combinations DOES NOT crash the app (key press combination `Ctrl + C: to copy to clipboard`, `Ctrl + S: save`, ecc.).

**Comment by mmahmoudian** [comment]
@MarkGotLasagna do you mean that even for drawing an arrow it crashes? or is it only for when you copy or save the screenshot?

**Comment by hissssst** [comment]
@MarkGotLasagna this issue is not about app crash
